### PR TITLE
enh: Fortran backend to visit `Print` arguments

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes https://github.com/lfortran/lfortran/issues/5221

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -1402,7 +1402,7 @@ RUN(NAME abs_03 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc wasm c fortran)
 RUN(NAME sqrt_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME sqrt_02 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME sin_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran)
-RUN(NAME sin_02 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran)
+RUN(NAME sin_02 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # fails on Mac
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:1405]: All these do not work on Mac

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -1537,10 +1537,10 @@ jobs:
...
         run: |
             cd integration_tests
-            ./run_tests.py -b fortran -j1
-            ./run_tests.py -b fortran -f -j1
-            ./run_tests.py -b fortran -j1 --experimental-simplifier
-            ./run_tests.py -b fortran -f -j1 --experimental-simplifier
+            ./run_tests.py -b fortran
+            ./run_tests.py -b fortran -f
+            ./run_tests.py -b fortran --experimental-simplifier
+            ./run_tests.py -b fortran -f --experimental-simplifier
```
Comment (Pranavchiku) [inline .github/workflows/CI.yml:1543]: Not sure why we had `-j1`, this slows CI and we all know how to catch which case got failed hence removed it

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -1585,8 +1585,8 @@ jobs:
         shell: bash -e -l {0}
         run: |
             cd integration_tests
-            ./run_tests.py -b mlir -j1
-            ./run_tests.py -b mlir -j1 --experimental-simplifier
+            ./run_tests.py -b mlir
+            ./run_tests.py -b mlir --experimental-simplifier
```
Comment (Pranavchiku) [inline .github/workflows/CI.yml:1589]: refer comment above

**Comment by Pranavchiku** [comment]
I don't think this failure has anything to do with this PR, this is exact same error I get on `sin_02` with latest master. CI: https://github.com/lfortran/lfortran/actions/runs/11626738999/job/32378899730?pr=5223

File: .github/workflows/CI.yml (hunk 2 above)
Comment (certik) [reply .github/workflows/CI.yml:1543]: I think this change caused https://github.com/lfortran/lfortran/pull/5223#issuecomment-2451754457.

Comment (certik) [reply .github/workflows/CI.yml:1543]: Remove this change from this PR, send it as a separate PR.

**Comment by Pranavchiku** [comment]
I opened https://github.com/lfortran/lfortran/issues/5227 for https://github.com/lfortran/lfortran/pull/5223#issuecomment-2451754457

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
I don't see any failure in master? https://github.com/lfortran/lfortran/actions?query=branch%3Amain

**Comment by Pranavchiku** [comment]
The failure comes when we run integration tests without -j1

**Comment by certik** [comment]
I see.

**Pull request merged**